lib/vector-core/src/lib.rs

```diff
@@ -8,9 +8,22 @@
 //! This library was extracted from the top-level project package, discussed in
 //! RFC 7027.
 
-#![deny(missing_docs)]
-#![deny(clippy::all)]
-#![deny(clippy::pedantic)]
+#![allow(clippy::float_cmp)]
+#![allow(clippy::approx_constant)]
```

Comment:
I'd like to see the `deny`s make it back in, but that's not really the scope of this PR.

Reply:
I would too. I'll make a follow-up issue. 

I had originally thought to make quality of life changes in this PR but it really ballooned the work out.